cartoframes/dataset.py

```diff
@@ -105,6 +107,29 @@ def set_geodataframe(self, gdf):
     def get_table_name(self):
         return self._table_name
 
+    def get_dataset_info(self):
+        if self._is_saved_in_carto:
+            if self._dataset_info is None:
+                self._dataset_info = self._get_dataset_info()
+
+            return self._dataset_info
+        else:
+            raise CartoException('Your data is not synchronized with CARTO.'
```

Comment:
Maybe we could move this raise to the beginning of the method as:

```python
if not self._is_saved_in_carto:
    raise ...

```

this way we can remove the `if... else`

Reply:
:+1: